apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/conf/dynamic/ConfigurationDiscoveryService.java

```diff
@@ -168,6 +172,30 @@ public void handleConfigurationDiscoveryCommand(ConfigurationDiscoveryCommand co
         LOGGER.trace("Current configurations after the sync, configurations:{}", register.toString());
     }
 
+    /**
+     * Read the registered dynamic configuration, compare it with the dynamic configuration information returned by the
+     * service, and complete the dynamic configuration that has been deleted on the OAP.
+     *
+     * @param configurationDiscoveryCommand Describe dynamic configuration information
+     * @return Adapted dynamic configuration information
+     */
+    private List<KeyStringValuePair> readConfig(ConfigurationDiscoveryCommand configurationDiscoveryCommand) {
+        Map<String, KeyStringValuePair> commandConfigs = configurationDiscoveryCommand.getConfig()
+                                                                                      .stream()
+                                                                                      .collect(Collectors.toMap(
+                                                                                          KeyStringValuePair::getKey,
+                                                                                          Function.identity()
+                                                                                      ));
+        List<KeyStringValuePair> configList = Lists.newArrayList();
+        for (final String name : register.keys()) {
```

Comment:
So you are building the list with all keys, right? That is what we missed today.

Reply:
Yes, sorry, I didn't think about it before.